### ISSUE TITLE
A VERY important suggestion that might help lots of people.

### ISSUE THREAD

**Issue by NumworksFan** [numworks/epsilon#2161]
Hi, devs of Numworks. Im a massive fan of your work, but I have a suggestion. I bought this calculator to help me do Algebra 2, for example, doing polar equations or long division with things like f(x)= 2x^3+5x^2-x^5+9 divided by x-2 or whatever. That type of stuff. It would be really nice if a update for doing Algebra 2 stuff would be made. Thanks!

Sincerely, a Numworks fan.

**Comment by jrkimber** [comment]
Hi,
It sounds like you're describing Computer Algebra (CAS) capablities. The Numworks doesn't do this as such calculators are banned in most exams.

Thanks,
James

**Comment by gentleAmateur** [comment]
> Hi, It sounds like you're describing Computer Algebra (CAS) capablities. The Numworks doesn't do this as such calculators are banned in most exams.
> 
> Thanks, James

That’s what exam mode is for. If CAS is not allowed, then you disable it during exams. And this is also achieved (rather clunkily, unfortunately) via χCAS. 

Which, ultimately, is what I suggest you look into @NumworksFan, and if/when Numworks considers adding CAS functionality then you may prefer other third-party applications. :) Just be advised that due to inane limitations to third-party apps it will unfortunately be a bit of a grueling experience.

**Comment by adriweb** [comment]
IIRC, some exams do not accept CAS-capable calculators at all even if it
can be disabled by some exam mode settings. That's why the TI-Nspire CX II
models are still split by CAS/non-CAS even if the CAS ones can disable it,
and why the HP Prime isn't allowed everywhere.

**Comment by jrkimber** [comment]
This is correct. Exam mode is not sufficient for many exams.That aside, as a teacher, I wouldn't really want students getting used to using CAS and then it disappearing in exams. They need to be using the same thing they'll be using in the exam. On 28 Feb 2024 19:40, Adrien Bertrand ***@***.***> wrote:
IIRC, some exams do not accept CAS-capable calculators at all even if it
can be disabled by some exam mode settings. That's why the TI-Nspire CX II
models are still split by CAS/non-CAS even if the CAS ones can disable it,
and why the HP Prime isn't allowed everywhere.

**Comment by gentleAmateur** [comment]
If that is true (I don’t really pay attention to what is and isn’t allowed in exams, my learning environment isn’t artificially restrictive) then it only reinforces my opinion that we would all be served better if Numworks had an official “pro” or “hacker” version of the calculator that was not for exams but for us who want powerful, extensible software with complete freedom and allows for native apps such as CAS that would be frowned upon in an exam setting.

**Comment by mobluse** [comment]
I adapted a polynomial division program to NumWorks' version of MicroPython:
https://my.numworks.com/python/mobluse/polydiv

There are also a lot of polynomial division programs for other graphing calculators e.g. TI-83 with only BASIC that could be ported to Python:
http://tibasicdev.wikidot.com/polynomial-division
https://www.ticalc.org/archives/files/fileinfo/425/42503.html
https://www.ticalc.org/archives/files/fileinfo/262/26200.html

I also made a version for ZX81 with 8 KiB ROM and 1 KiB RAM (1981):
https://github.com/mobluse/zx81-mob/blob/master/mob-i-l/polydiv1k.bas
